framework/include/constraints/MortarConstraintBase.h

```diff
@@ -108,11 +108,10 @@ class MortarConstraintBase : public Constraint,
   void zeroInactiveLMDofs(const std::unordered_set<const Node *> & inactive_lm_nodes,
                           const std::unordered_set<const Elem *> & inactive_lm_elems);
 
-private:
+protected:
   /// Reference to the finite element problem
   FEProblemBase & _fe_problem;
 
-protected:
```

Comment:
@lindsayad acceptable?

Reply:
Do you need the non-const reference? If not, you could port what I've done here: https://github.com/idaholab/moose/pull/19266/files#diff-7345a9b3d222670aa12c4f58f967283bd34131dad796dbea8237031038ce84b4R111

Reply:
`const` ref is fine, that works.